docs/meshstack.customer-group-sync.md

```diff
@@ -4,3 +4,206 @@ title: meshCustomer Group Synchronisation
 ---
 
 meshStack supports importing users and groups into meshStack from a source that supports the LDAP protocol, for example, Microsoft Active Directory. The [`Simple Paged Results Control`](https://www.ietf.org/rfc/rfc2696.txt) is used to do a paginated LDAP query to fetch the users and groups. Because of this, it is required that the LDAP server supports the `Simple Paged Results Control`. Other than importing users and groups, meshStack also has the capability to assign a [customer role](./meshcloud.customer.md#assign-meshcustomer-roles) to the imported group. The entities read via LDAP are imported to meshStack via the [meshObject import API](./meshstack.api.md#meshobject-api).
+
+## Configuration Reference
+
+This section describes how to configure the LDAP group synchronization in meshStack.
+For easier reference the following sections break down the configuration model in multiple parts. The union of these
+defines the full configuration model.
+
+The basic parameters required to configure the connection to LDAP are as follows.
+
+<!--snippet:mesh.identityconnector.ldap.core#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+let ConnectionConfiguration =
+    {-
+      url: The LDAP connection URL which would be of the form ldap://example.com:389
+      base:
+          The base from which searches should be performed, which would be of the form "dc=meshcloud,dc=io"
+      username: The username of the user who can connect to the LDAP source to perform searches
+      password: The password of the user above
+    -}
+      { url : Text, base : Text, username : Text, password : Secret }
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+The configuration related to the collection of data via LDAP is as follows.
+
+<!--snippet:mesh.identityconnector.ldap.collector#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+{-
+  pageSize: The size of a single page that will be returned as a result of an LDAP search
+  group/person:
+    a configuration of type LDAPFilter which will contain the information needed to search the LDAP source
```

Comment:
```suggestion
    a configuration of type LDAP filter which will contain the information needed to search the LDAP source
```

Reply:
I think Pububdu called it that way because its an actual type but I am not sure if we should bother the readers here with actual types. Maybe go the route that we only mention its an so called LDAP filter and here you have the info how to setup such a filter for our system.

Reply:
Yes. It is an actual type and it's structure is shown right below.

Reply:
@tfelix agreed